### ISSUE TITLE
formatter mode-result error when trying to format/send mail from mu4e

### ISSUE THREAD

**Issue by jonathanwilner** [hlissner/doom-emacs#3270]
**What did you expect to happen?**

Expected mail to be formatted & sent via smtpmail without error. 

**What actually happened?**'

HTML format generated, but there's an error pre-SMTP sending. 

**Additional details:**

<details><pre>

```
Debugger entered--Lisp error: (wrong-number-of-arguments (formatter mode-result) 0)
  signal(wrong-number-of-arguments ((formatter mode-result) 0))
  format-all-buffer--from-hook()
  run-hooks(before-save-hook)
  basic-save-buffer(nil)
  save-buffer()
  #f(compiled-function () #<bytecode 0x1ff523745161>)()
  run-hooks(message-send-hook)
  message-send(nil)
  message-send-and-exit(nil)
  funcall-interactively(message-send-and-exit nil)
  call-interactively(message-send-and-exit nil nil)
  command-execute(message-send-and-exit) 

```

</pre></details> ```

**System information:**
<details><pre>

```
emacs   version    27.0.91
        features   RSVG DBUS GLIB NOTIFY KQUEUE ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS MODULES THREADS JSON PDUMPER LCMS2 GMP
        build      Apr 20, 2020
        buildopts  (--with-modules --without-makeinfo)
        windowsys  mac
        daemonp    server-running
doom    version    2.0.9
        build      HEAD -> develop c5b561f79 2020-06-01 05:51:26 -0400
        dir        ~/.doom.d/
system  type       darwin
        config     x86_64-apple-darwin19.4.0
        shell      /bin/zsh
        uname      Darwin 19.5.0 Darwin Kernel Version 19.5.0: Thu Apr 30 18:25:59 PDT 2020; root:xnu-6153.121.1~7/RELEASE_X86_64 x86_64
        path       (bin/../bin ~/opt/anaconda3/bin ~/opt/anaconda3/condabin /usr/local/bin /usr/bin /bin /usr/sbin /sbin /Library/TeX/texbin /usr/local/share/dotnet ~/.dotnet/tools /Library/Apple/usr/bin /Library/Frameworks/Mono.framework/Versions/Current/Commands /usr/local/libexec/emacs/27.0.91/x86_64-apple-darwin19.4.0/)
config  envfile    envvar-file
        elc-files  0
        modules    (:completion company ivy :ui doom doom-dashboard hl-todo modeline nav-flash ophints (popup +all +defaults) treemacs unicode vc-gutter vi-tilde-fringe window-select workspaces zen :editor (evil +everywhere) file-templates fold (format +onsave) snippets :emacs (dired +icons) electric undo vc :term eshell vterm :checkers syntax spell :tools (eval +overlay) lookup lsp macos magit (pass +password-store +password-store-otp) pdf prodigy :lang data emacs-lisp (go +lsp) (javascript +lsp) latex markdown (org +dragndrop +gnuplot +habit +journal +jupyter +noter +pandoc +pdfview +pomodoro +present +roam) plantuml (python +lsp) sh (swift +lsp) web :email (mu4e +gmail) (notmuch +mail) :app twitter :config (default +bindings +smartparens))
        packages   ((slack :pin 61ff6fa71f32b58ce8cb9c5fc88b0f20b54bb0df) (ivy-emoji) (excorporate) (alert) (mu4e-alert) (virtual-auto-fill :recipe (:host github :repo luisgerhorst/virtual-auto-fill)) (org-super-agenda) (bookmarkplus :recipe (:host github :repo emacsmirror/bookmark-plus)) (counselweb :recipe (:host github :repo mnewt/counsel-web)) (org-web-tools :recipe (:host github :repo alphapapa/org-web-tools)) (ob-mermaid :recipe (:host github :repo arnm/ob-mermaid)) (modus-operandi-theme) (modus-vivendi-theme) (org-graph-view :recipe (:host github :repo alphapapa/org-graph-view)) (ox-jira) (calfw) (calfw-org) (calfw-cal) (pandoc-mode) (atomic-chrome) (yequake) (magit-delta))
        unpin      (n/a)
        elpa       (n/a)
```
</details>
</pre>

**Comment by JakDar** [comment]
Simmilar thigs happens for me when trying to format code in org src blocks using `org-babel-do-key-sequence-in-edit-buffer` and `SPC c f` .  Formatting regular files which use mode related to used src block babel language (e.g. python) works.
<details>

```
Debugger entered--Lisp error: (wrong-number-of-arguments (formatter mode-result) 0)
  signal(wrong-number-of-arguments ((formatter mode-result) 0))
  format-all-buffer()
  funcall-interactively(format-all-buffer)
  call-interactively(format-all-buffer)
  +format/buffer()
  funcall-interactively(+format/buffer)
  call-interactively(+format/buffer)
  +format/region-or-buffer()
  funcall-interactively(+format/region-or-buffer)
  call-interactively(+format/region-or-buffer)
  org-babel-do-key-sequence-in-edit-buffer(" cf")
  funcall-interactively(org-babel-do-key-sequence-in-edit-buffer " cf")
  call-interactively(org-babel-do-key-sequence-in-edit-buffer nil nil)
  command-execute(org-babel-do-key-sequence-in-edit-buffer)

```


</details>

**Comment by hlissner** [comment]
This issue should've been fixed a week ago in 991a205ebd43, a day after you posted, so I'll consider this resolved. Let me know if that isn't the case and I'll reopen it.
